### PR TITLE
Migrating to Github Actions

### PR DESCRIPTION
**Pull request by anandtiwary**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Migrating to Github actions

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[X] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/ngneat/spectator/issues/410

## What is the new behavior?
Added a github actions workflow file. Did not remove Travis.yml yet.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by anandtiwary** [comment]
Example from forked repo: https://github.com/anandtiwary/spectator/runs/2144612854 
Not sure why it didn't run automatically on this PR.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: build-and-test
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+jobs:
```
Comment (NetanelBasal) [inline .github/workflows/build-and-test.yml:7]: Any specific reason you chose multiple jobs over one job with multiple steps?

Comment (anandtiwary) [reply .github/workflows/build-and-test.yml:7]: Yeah.. Build and test jobs can execute in parallel. Verification will take less time.

**Comment by NetanelBasal** [comment]
Can you please delete travis?

**Comment by anandtiwary** [comment]
done

**Pull request merged**